spec/requests/api/v1/users/create_spec.rb

```diff
@@ -0,0 +1,122 @@
+  let(:internal_server_error) { 500 }
+
+  describe 'POST create' do
+    let(:email)                 { 'test@test.com' }
+    let(:password)              { '12345678' }
+    let(:password_confirmation) { '12345678' }
+    let(:name)                  { 'Sebastian' }
+    let(:gender)                { 'male' }
+
+    let(:params) do
+      {
+        user: {
+          email: email,
+          password: password,
+          password_confirmation: password_confirmation,
+          name: name,
+          gender: gender
+        }
+      }
+    end
+
+    it 'returns a succesful response' do
+      post user_registration_path, params: params, as: :json
+
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'creates the user' do
+      expect do
+        post user_registration_path, params: params, as: :json
+      end.to change(User, :count).by(1)
+    end
+
+    it 'returns the user' do
+      post user_registration_path, params: params, as: :json
+
+      expect(json[:user][:id]).to eq(user.id)
+      expect(json[:user][:email]).to eq(user.email)
+      expect(json[:user][:name]).to eq(user.name)
+      expect(json[:user][:gender]).to eq(user.gender)
+    end
+
+    context 'when email is not correct' do
+      let(:email) { 'invalid_email' }
+
+      it 'does not create a user' do
+        expect do
+          post user_registration_path, params: params, as: :json
+        end.not_to change { User.count }
+      end
+
+      it 'does not return a successful response' do
+        post user_registration_path, params: params, as: :json
+        expect(response.status).to eq(failed_response)
+      end
+    end
+
+    context 'when the password is incorrect' do
+      let(:password) { 'short' }
+      let(:password_confirmation) { 'short' }
+      let(:new_user)              { User.find_by_email(email) }
+
+      it 'does not create a user' do
+        post user_registration_path, params: params, as: :json
+
+        expect(new_user).to be_nil
+      end
+
+      it 'does not return a successful response' do
+        post user_registration_path, params: params, as: :json
+
+        expect(response.status).to eq(failed_response)
+      end
+    end
+
+    context 'when passwords do not match' do
+      let(:password)              { 'shouldmatch' }
+      let(:password_confirmation) { 'dontmatch' }
+      let(:new_user)              { User.find_by_email(email) }
+
+      it 'does not create a user' do
+        post user_registration_path, params: params, as: :json
+        expect(new_user).to be_nil
+      end
+
+      it 'does not return a successful response' do
+        post user_registration_path, params: params, as: :json
+        expect(response.status).to eq(failed_response)
+      end
+    end
+
+    context 'when name is blank' do
+      let(:name) { '' }
+      let(:new_user)              { User.find_by_email(email) }
+
+      it 'does not create a user' do
+        post user_registration_path, params: params, as: :json
```

Comment:
You should use `shared_examples` to reuse code.
https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples